src/main/java/com/hubspot/jinjava/lib/fn/Functions.java

```diff
@@ -145,4 +148,59 @@ public static int movePointerToJustBeforeLastWord(int pointer, String s) {
     return pointer + 1;
   }
 
+  @JinjavaDoc(value = "<p>Return a list containing an arithmetic progression of integers.</p>" +
+      "<p>With one parameter, range will return a list from 0 up to or down to the value. " +
+      " With two parameters, the range will start at the first value and end at the second value. " +
+      " The third parameter specifies the step increment.</p> <p>All values can be negative.</p>" +
+      "<p>Ranges can generate a maximum of " + RANGE_LIMIT + " values.</p>",
+      params = {
+          @JinjavaParam(value = "start", type = "number", defaultValue = "0"),
+          @JinjavaParam(value = "end", type = "number"),
+          @JinjavaParam(value = "step", type = "number", defaultValue = "1")})
+  public static List<Integer> range(Object arg1, Object... args) {
+
+    List<Integer> result = new ArrayList<>();
+
+    int start = 0;
+    int end;
+    int step = 1;
+
+    switch (args.length) {
+      case 0:
+        end = Integer.parseInt(arg1.toString());
+        break;
+      case 1:
+        start = Integer.parseInt(arg1.toString());
+        end = Integer.parseInt(args[0].toString());
+        break;
+      default:
+        start = Integer.parseInt(arg1.toString());
+        end = Integer.parseInt(args[0].toString());
+        step = Integer.parseInt(args[1].toString());
+    }
+
+    if (start < end) {
+      for (int i = start; i < end; i += step) {
+        if (result.size() >= RANGE_LIMIT) {
+          break;
+        }
+        result.add(i);
+      }
+    } else {
+
+      // make sure we're decrementing
```

Comment:
This can be moved to a top-level `if (start < end && step > 0) { step = step * -1;}` and you won't have to copy-paste the for loop.

That being said, I think that trying to interpret the user's intention by fixing their buggy inputs is generally a bad idea. Better to return an empty list and let them figure out what they did wrong.